### ISSUE TITLE
Please implement GBM_BO_IMPORT_EGL_IMAGE

### ISSUE THREAD

**Issue by shadeslayer** [rockchip-linux/libmali#42]
Hi
It'd be great if you could implement GBM_BO_IMPORT_EGL_IMAGE in libmali. This would allow us to use the interface in eglamor in Xserver without having to write a workaround for libmali.

**Comment by Apokalypzx** [comment]
I agree. However, I'm pretty sure that ARMs DDK (the source code rockchip modifies and compiles to build this binary) doesn't work that way. It is my understanding that only the newer versions of this driver would have the new api calls in gbm.h. Since we're stuck on the same kernelspace/userspace version (example: r14 for T860), there's no way to write in the new GBM calls without messing with the code too much (which is also probably against ARM's licenses for the DDK.)

In other words, this probably won't happen without a version bump. So Rockchip, please build against the DDK with feature parity to what we need to run glamor as it now supports GLES. Also, get it to work with the new gl vendor dispatch library, that will make integration so much easier with mesa on the same system.

**Comment by shadeslayer** [comment]
> I agree. However, I'm pretty sure that ARMs DDK (the source code rockchip modifies and compiles to build this binary) doesn't work that way. It is my understanding that only the newer versions of this driver would have the new api calls in gbm.h. Since we're stuck on the same kernelspace/userspace version 
(example: r14 for T860), there's no way to write in the new GBM calls without messing with the code too much (which is also probably against ARM's licenses for the DDK.)

True, however, since almost all distributions will end up using gbm.h from mesa instead of the one from libMali, as long as the call has been implemented in libMali, everything will work.

> 
> In other words, this probably won't happen without a version bump. So Rockchip, please build against the DDK with feature parity to what we need to run glamor as it now supports GLES. Also, get it to work with the new gl vendor dispatch library, that will make integration so much easier with mesa on the same system.

I presume you're talking about libglvnd? I haven't really looked at it much tbh, but I've heard it being mentioned a few times as the right way to move forward.
